xsens-tf-bridge/src/xsens-tf-bridge.cpp

```diff
@@ -0,0 +1,262 @@
+        tf.transforms[i].transform.translation.y = xsensData->segmentsData[i].position.c2;
+        tf.transforms[i].transform.translation.z = xsensData->segmentsData[i].position.c3;
+        tf.transforms[i].transform.rotation.x = xsensData->segmentsData[i].orientation.c2;
+        tf.transforms[i].transform.rotation.y = xsensData->segmentsData[i].orientation.c3;
+        tf.transforms[i].transform.rotation.z = xsensData->segmentsData[i].orientation.c4;
+        tf.transforms[i].transform.rotation.w = xsensData->segmentsData[i].orientation.c1;
+        }
+        
+        for (i=(trueSegmentCount-1);i<segmentsCount;i++){
+        tf.transforms[i].header.seq   = i;
+        tf.transforms[i].header.stamp = normalizeSecNSec(xsensData->time);
+        tf.transforms[i].transform.translation.x = 0;
+        tf.transforms[i].transform.translation.y = 0;
+        tf.transforms[i].transform.translation.z = 0;
+        tf.transforms[i].transform.rotation.x = 0;
+        tf.transforms[i].transform.rotation.y = 0;
+        tf.transforms[i].transform.rotation.z = 0;
+        tf.transforms[i].transform.rotation.w = 1;
+        }
+
+        publisher_tf.write(tf);
+
+        i++;
+        for (i=0;i<segmentsCount;i++){
+        cout << tf.transforms[i].child_frame_id << endl;
+        cout << "position [" << tf.transforms[i].transform.translation.x << " " << tf.transforms[i].transform.translation.x << " " << tf.transforms[i].transform.translation.x << "]" << endl;
+        cout << "orientation [" << tf.transforms[i].transform.rotation.w << " " << tf.transforms[i].transform.rotation.x << " " << tf.transforms[i].transform.rotation.y << " " << tf.transforms[i].transform.rotation.z << "]" << endl;
+        }
+        xsensDataPort.write();
+        return true;
+    }
+    // Message handler. Just echo all received messages.
+    bool respond(const Bottle& command, Bottle& reply)
+    {
+        cout << "Got something, echo is on" << endl;
+        if (command.get(0).asString() == "quit")
+            return false;
+        else
+            reply = command;
+        return true;
+    }
+    // Configure function. Receive a previously initialized
+    // resource finder object. Use it to configure your module.
+    // If you are migrating from the old module, this is the function
+    // equivalent to the "open" method.
+    bool configure(yarp::os::ResourceFinder &rf)
+    {     
+        if (!publisher_tf.topic("/tf")) {
+        cerr<< "Failed to create publisher to /tf\n";
+        return -1;
+        }
+        
+        xsensDataPort.open("/xsens_data_reader");
+        xsensServerName = "/xsens/frames:o"; // >>yarp conf 10.255.36.7 10000 >> yarp rpc /xsens/cmd:i >>yarp read ... /xsens/frames:o
+        if (!Network::connect("/xsens_data_writer","/xsens_data_reader"))
+        {
+           std::cout << "Error! Could not connect to server " << xsensServerName << std::endl;
+           return -1;
```

Comment:
`bool` is not a number. 
The `int` `-1` is automatically casted to `bool` and given the rules in C++ for converting `int` to `bool` is that `0` is casted to `false` and everything else is casted to `true`, you are actually returning `true` in this function!